### PR TITLE
Use culture invariant representation for preview value

### PR DESCRIPTION
**Pull request by ke-yu**
This is to fix [defect magn-6714] (http://adsk-oss.myjetbrains.com/youtrack/issue/MAGN-6714) that on German System the decimals are displayed as comma for when mouse moved to preview. 

By default if an object is a formattable object, `Object.ToString()` will use culture dependent format to give the string representation of this object. We should format it in culture invariant way.

@ikeough please review the change. Thanks.

### CODE REVIEW

**Comment by RodRecker** [comment]
Let's have someone else look at this while Ian is on vacation.

**Comment by ke-yu** [comment]
@Benglin please help to review the change. Thanks.

**Comment by ke-yu** [comment]
@Randy-Ma please help to review the change

File: src/Engine/ProtoCore/Reflection/MirrorData.cs [hunk 1]
```diff
@@ -205,24 +207,26 @@ public string StringData
...
+                        return Data.ToString().ToLower();
+                    }
+                    else if (Data is IFormattable)
+                    {
+                        // Object.ToString() by default will use the current 
+                        // culture to do formatting. For example, Double.ToString()
+                        // https://msdn.microsoft.com/en-us/library/3hfd35ad(v=vs.110).aspx
+                        // We should always use invariant culture format for formattable 
+                        // object.
+                        return (Data as IFormattable).ToString(null, CultureInfo.InvariantCulture);
```
Comment (Randy-Ma) [inline src/Engine/ProtoCore/Reflection/MirrorData.cs]: We can directly use ToString(CultureInfo.InvariantCulture)

File: src/Engine/ProtoCore/Reflection/MirrorData.cs [hunk 2]
```diff
@@ -205,24 +207,26 @@ public string StringData
...
                     {
                         return "null";
                     }
+                    else if (Data is bool)
+                    {
+                        return Data.ToString().ToLower();
+                    }
+                    else if (Data is IFormattable)
+                    {
+                        // Object.ToString() by default will use the current
```
Comment (Randy-Ma) [inline src/Engine/ProtoCore/Reflection/MirrorData.cs]: We can only keep the first sentence for the comments.

**Comment by Randy-Ma** [comment]
@ke-yu 
Thanks a lot! Other than the two comments, LGTM!

**Pull request merged**